src/Raven.Pal/src/posix/pager.c

```diff
@@ -598,4 +598,102 @@ int32_t rvn_write_invalid_setup(
 {
     *detailed_error_code = ENOTSUP;
     return FAIL_INVALID_HANDLE;
-}
\ No newline at end of file
+}
+
+int32_t rvn_write_mmap32(
+    void *handle,
+    struct page_to_write *buffers,
+    int32_t count,
+    int32_t *detailed_error_code)
+{
+    int32_t rc = SUCCESS;
+
+    void* mapped_view = NULL;
+    int64_t mapped_start_page = -1;  // Start page of currently mapped region
+    int64_t mapped_page_count = 0;   // Number of pages in currently mapped region
+
+    for (int32_t i = 0; i < count; i++)
+    {
+        int64_t page_number = buffers[i].page_num;
+        int64_t number_of_pages = buffers[i].count_of_pages;
+
+        // Check if the current write fits in the currently mapped region
+        bool needs_remap = (mapped_view == NULL) ||
+                          (page_number < mapped_start_page) ||
+                          (page_number + number_of_pages > mapped_start_page + mapped_page_count);
+
+        if (needs_remap)
+        {
+            // Flush and unmap the previous region if it exists
+            if (mapped_view != NULL)
+            {
+                int64_t mapped_size = mapped_page_count * VORON_PAGE_SIZE;
+
+                if (msync(mapped_view, (size_t)mapped_size, MS_SYNC))
```

Comment:
Why are we calling `msync` here? given: https://github.com/ravendb/ravendb/pull/21565/files#r2431950537